tsconfig.json

```diff
@@ -4,7 +4,7 @@
     "declaration": true,
     "esModuleInterop": true,
     "inlineSources": true,
-    "lib": ["dom", "es2017"],
+    "lib": ["dom", "es2018"],
```

Comment:
Are the changes to this file mandatory? I can't see any new browser APIs being used in the code you've added...

Reply:
The change to es2018 is required because the code uses Promise.prototype.finally() in src/timeout.ts:8. This method was introduced in ES2018/ES2019 spec.

Without this change, the TypeScript compiler throws

'error TS2550: Property 'finally' does not exist on type 'Promise<unknown>'. 
Do you need to change your target library? Try changing the 'lib' compiler option to 'es2018' or later.'

The finally() method is used in the timeout utility to ensure cleanup (clearing the timeout) happens regardless of whether the promise resolves or rejects. This is the only ES2018 feature being used, but it's essential for the proper cleanup logic.